internal/api/impl/auth/oidc.go

```diff
@@ -367,8 +371,10 @@ func retrieveDeviceAccessToken(ctx context.Context, relyingParty RelyingPartyWit
 func (e *oIDCEndpoint) retrieveClientCredentialsToken(ctx context.Context, clientID string, clientSecret string) (*oauth2.Token, error) {
 	// Create a new client credentials request
 	req := &oidc.ClientCredentialsRequest{
+		GrantType:    oidc.GrantTypeClientCredentials,
 		ClientID:     clientID,
 		ClientSecret: clientSecret,
+		Scope:        e.relyingParty.OAuthConfig().Scopes,
```

Comment:
TODO in azure, the scope is different for client credentials flow. This needs to be configurable.